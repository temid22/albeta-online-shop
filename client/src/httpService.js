import axios from 'axios';

// development environment baseUrl
axios.defaults.baseURL = 'http://localhost:5000/api/';

// production environment baseUrl
// axios.defaults.baseURL = "https://react-node-u6i6.onrender.com/api/";

const user = JSON.parse(localStorage.getItem('user'));

const TOKEN = user?.accessToken;

// axios request for api calls without token/headers configuration
export const generalRequest = axios.create({});

// axios request for api calls with token/headers configuration
export const userRequest = axios.create({
  headers: { token: `Bearer ${TOKEN}` },
});
//
