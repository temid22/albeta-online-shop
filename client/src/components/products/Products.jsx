import React, { useEffect, useState } from 'react';
import Image from '../../images/product.jpg';
import './products.css';
import { generalRequest, userRequest } from '../../httpService';

const Products = () => {
  // products state
  const [products, setProducts] = useState([]);

  // get user from Local storage
  const user = JSON.parse(localStorage.getItem('user'));

  let cart = [];
  // add product to cart
  const addToCart = async (product) => {
    cart.push(product);
    try {
      const { data } = await userRequest.post('/cart', {
        id: user._id, // userId
        product: [
          {
            productId: product._id, //product Id
          },
        ],
        //price
        amount: product.price,
        total: product.price * 1,
      });
      console.log(data, 'added to cart ');

      localStorage.setItem('cart', JSON.stringify(cart));
    } catch (error) {
      console.log(error);
    }
  };

  // fetch products when page renders
  useEffect(() => {
    const getProducts = async () => {
      try {
        const { data } = await generalRequest.get('/product');
        console.log(data, 'Products');

        setProducts(data);
      } catch (error) {
        console.log(error);
      }
    };

    getProducts(); // return the get products function
  }, []);

  // add product to cart

  return (
    <div>
      <div className='productContainer'>
        <div className='productTitle'>Our Products</div>
        <div className='productWrapper'>
          {products.length > 0 ? (
            products?.map((product) => (
              <div className='prodCard' key={product._id}>
                <div className='prodImg'>
                  <img src={product.img} alt='Product Image' />
                </div>
                <div className='prodTitle'>{product.title}</div>
                <div className='prodDesc'>{product.desc}</div>
                <div className='prodColor'>{product.color}</div>
                <div className='prodPrice'>Â£{product.price}</div>
                <button onClick={() => addToCart(product)} className='cartBtn'>
                  Add To Cart
                </button>
              </div>
            ))
          ) : (
            <div>Loading...</div>
          )}
        </div>
      </div>
    </div>
  );
};

export default Products;
