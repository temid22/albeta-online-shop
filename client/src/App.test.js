//import { render, screen, cleanup } from "@testing-library/react";
// import dependencies
import renderer from "react-test-renderer";
//import { BrowserRouter } from "react-router-dom";
//import App from "./components/products/Products";
import { generalRequest, userRequest } from "./httpService.js";
import Products from "./components/products/Products";

// snapshot test
test("Product component snapshot test", function () {
  let tree = renderer.create(<Products />).toJSON(); // get json object representing the app
  console.log(tree);
  expect(tree).toMatchSnapshot(); // compare to snapshot
});

// unit test
test("test general request function", async function () {
  const response = await generalRequest.get("/product");

  //console.log(response);

  expect(response.status).toEqual(200); // compare result status to be 200
});

test("test general user request function", async function () {
  const response = await userRequest.get("/product");

  //console.log(response);

  expect(response.status).toEqual(200); // compare result status to be 200
});
