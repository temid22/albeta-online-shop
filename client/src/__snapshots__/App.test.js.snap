// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product component snapshot test 1`] = `
<div>
  <div
    className="productContainer"
  >
    <div
      className="productTitle"
    >
      Our Products
    </div>
    <div
      className="productWrapper"
    />
  </div>
</div>
`;

exports[`renders correctly 1`] = `
<div>
  <div
    className="productContainer"
  >
    <div
      className="productTitle"
    >
      Our Products
    </div>
    <div
      className="productWrapper"
    />
  </div>
</div>
`;
